### PR TITLE
compiler: Avoid allocating Bundles on the host if transient

### PR DESCRIPTION
**Pull request by FabioLuporini**
Impacted test tweaked in PRO

### CODE REVIEW

File: devito/passes/iet/definitions.py [hunk 1]
```diff
@@ -214,6 +216,38 @@ def _alloc_mapped_array_on_high_bw_mem(self, site, obj, storage, *args):
 
         storage.update(obj, site, allocs=alloc, frees=free, efuncs=(efunc0, efunc1))
 
+    def _alloc_bundle_struct_on_high_bw_mem(self, site, obj, storage):
```
Comment (EdCaunt) [inline devito/passes/iet/definitions.py:219]: Nitpick: some comments would help reduce the bus factor of this code

File: devito/passes/iet/definitions.py [hunk 2]
```diff
@@ -340,9 +374,22 @@ def place_definitions(self, iet, globs=None, **kwargs):
         for i in FindSymbols().visit(iet):
             if i in defines:
                 continue
+
             elif i.is_LocalObject:
                 self._alloc_object_on_low_lat_mem(iet, i, storage)
-            elif i.is_Array or i.is_Bundle:
+
+            elif i.is_Bundle:
```
Comment (EdCaunt) [inline devito/passes/iet/definitions.py:381]: Is there any way to reduce the "death by conditionals" here?

Comment (FabioLuporini) [reply devito/passes/iet/definitions.py:381]: After spending a couple of hours on this very matter this morning, I'm inclined to say that it's pretty challenging to come up with one. The problem is that there's a combinatorial explosion of possibilities (luckily with small exponent), and all those `if`s implement such space

File: devito/passes/iet/definitions.py (hunk 1 above)
Comment (FabioLuporini) [reply devito/passes/iet/definitions.py:219]: I agree, if it turns green I will add it to the to-be-rebased #2500

File: devito/passes/iet/definitions.py (hunk 2 above)
Comment (EdCaunt) [reply devito/passes/iet/definitions.py:381]: I suppose it's all attributes too, which makes it even more challenging without obfuscating what's going on

**Review by EdCaunt — APPROVED** [review]

File: devito/passes/iet/definitions.py (hunk 2 above)
Comment (FabioLuporini) [reply devito/passes/iet/definitions.py:381]: if the implicit question was "can you use singledispatch", yeah, you got it right

**Pull request merged**